modules/diskq/qdisk.c

```diff
@@ -511,6 +527,22 @@ _load_non_reliable_queues(QDisk *self, GQueue *qout, GQueue *qbacklog, GQueue *q
   return TRUE;
 }
 
+static gboolean
+_load_non_reliable_queues(QDisk *self, GQueue *qout, GQueue *qbacklog, GQueue *qoverflow)
```

Comment:
I like this refactor, but I think we can resolve the code repeating problem without using macros and string mangling.
How about creating structs which contain the queue, and the additional members (ofs, len, count), then we create
an array out of these structures, on which we can freely iterate.

The error handling should be done outside of the iterator where we can access the names of the current index "qout", "qbacklog", etc.

Reply:
The main reason this macro complexity is involved is because I wanted to retain the original error message in case of corruption detection. By the way testdb also tests the error message. Though we believe with Gabor that it might worth breaking the error message for sake of reduced complexity.
With that, I believe I can simplify the macros considerably, and turn `_try_to_load_queue` into a function. I will try it out and get back with a new version.

Reply:
Thanks, I like the changes.